### ISSUE TITLE
Keep Database after restart

### ISSUE THREAD

**Issue by MarcelCoding** [ONLYOFFICE/DocumentServer#2787]
### This issue is unique.

- [X] I have used the [search tool](https://github.com/ONLYOFFICE/DocumentServer/issues?q=) and did not find an issue describing my idea.

### Your idea.

> If nextcloud is down, or the changes can otherwise not be saved back to nextcloud (e.g. because of https://github.com/ONLYOFFICE/onlyoffice-nextcloud/issues/931), the onlyoffice database contains all the user data and should absolutely not be dropped on startup!
> ~ https://github.com/NixOS/nixpkgs/pull/319858#issue-2353763414

Not to drop the database is currently not possible:
> The only problem I could think of is the absence of database migrations because upstream is dropping everything on startup.
> ~ https://github.com/NixOS/nixpkgs/pull/319858#issuecomment-2180566587

I've also noticed that OnlyOffice displays in the bottom corner that all changes are saved. 
![image](https://github.com/ONLYOFFICE/DocumentServer/assets/34819524/fd9c820d-2643-4e19-a228-7889d1ed38ba)

The actual error when saving changes back to OnlyOffice only occurs when using the save button in the menu.
![image](https://github.com/ONLYOFFICE/DocumentServer/assets/34819524/0322e69e-42b3-4907-80c4-3026cc5b3d11)

It would be nice, that the user would be informed instantly if there are errors writing to the underlying storage provider. (e.g. Nextcloud)

**Comment by igwyd** [comment]
Hello @MarcelCoding, I think you need an [forcesave](https://api.onlyoffice.com/editors/command/forcesave) option, in Nextcloud it's named "Keep intermediate version when editing (forcesave)". This changes the behavior of the "save" button: if Nextcloud is down and you will press "save" then you will get message "An error occured while saving the file...".

**Comment by MarcelCoding** [comment]
Cool, that would be a solution. But I think the database should not be dropped anyways. A user and an admin is not the same person and maybe to resolve the problem (why files cannot be saved) a restart is necessary. Therefore the data should be kept only when restarting onlyoffice.

**Comment by MarcelCoding** [comment]
Maybe the default should be changed in nextcloud or at least the wording with the explicit explanation that this can happen if you don't enable the option.

**Comment by igwyd** [comment]
> Maybe the default should be changed in nextcloud or at least the wording with the explicit explanation that this can happen if you don't enable the option.

We have it in the connector [documentation](https://helpcenter.onlyoffice.com/integration/gettingstarted-nextcloud.aspx#Configure_block:~:text=Keep%20intermediate%20versions,editors%27%20cache%20automatically.).

> Cool, that would be a solution. But I think the database should not be dropped anyways. A user and an admin is not the same person and maybe to resolve the problem (why files cannot be saved) a restart is necessary. Therefore the data should be kept only when restarting onlyoffice.

Changes are deleted from the database after they are sent to the document server cache, for example, this will happen if the editor receives the [status](https://api.onlyoffice.com/editors/callback#status-descr) of a closed document and this is regardless of the availability of the storage. And then this cache will wait until the storage is turned on and the first user who called the document will receive changes from the cache.

**Comment by MarcelCoding** [comment]
> Changes are deleted from the database after they are sent to the document server cache, for example, this will happen if the editor receives the status of a closed document and this is regardless of the availability of the storage. And then this cache will wait until the storage is turned on and the first user who called the document will receive changes from the cache

Sadly this check is missing here: https://github.com/ONLYOFFICE/Docker-DocumentServer/blob/0d5588d03a7672b8274e93d9151f9449f3dd499c/run-document-server.sh#L490
And because of that the cache is cleared all the time onlyoffice restarts, even if the data is not already written to the storage...

**Comment by MarcelCoding** [comment]
https://github.com/ONLYOFFICE/onlyoffice-nextcloud/issues/931#issuecomment-2061870498 is probably the reason files count not be moved from the database to the cache. But the data should not be automatically cleared from the database if the creation of the cache failed.

**Comment by igwyd** [comment]
> That's interesting because the time I encountered data loss I've looked into all directories and there weren't an files.
Because there was a restart in between the data was deleted from the database and I was able to recover the data from database backups. Now the question is, why wasn't the data written to the files? I should have not been able to recover the data from the database days after is was written to the database ? https://github.com/ONLYOFFICE/Docker-DocumentServer/blob/0d5588d03a7672b8274e93d9151f9449f3dd499c/run-document-server.sh#L490
And because of that the cache is cleared all the time onlyoffice restarts, even if the data is not already written to the storage...

When the container is restarted, it starts [documentserver-prepare4shutdown.sh](https://github.com/ONLYOFFICE/Docker-DocumentServer/blob/0d5588d03a7672b8274e93d9151f9449f3dd499c/run-document-server.sh#L8), that forcibly disconnects all users from documents and sends all changes from the database to the cache and then to the storage (nextcloud in your case). This is done to avoid data loss when stopping or restarting the container. 

The entire cache `/var/lib/onlyoffice/documentserver/App_Data/cache/files/` will be deleted when the container is restarted, you need to mount it like [here](https://helpcenter.onlyoffice.com/installation/docs-community-install-docker.aspx#:~:text=To%20get%20access%20to%20your%20data%20located%20outside%20the%20container%2C%20you%20need%20to%20mount%20the%20volumes.%20It%20can%20be%20done%20by%20specifying%20the%20%2Dv%20option%20in%20the%20docker%20run%20command.) to avoid this.

**Comment by igwyd** [comment]
> But the data should not be automatically cleared from the database if the creation of the cache failed.
Agree, could you please describe the case of data loss in more detail so that I can repeat it?

**Comment by MarcelCoding** [comment]
The cache could probably not created, because x2t failed: https://github.com/ONLYOFFICE/onlyoffice-nextcloud/issues/931

(x2t is only failing for some document formats, not all)

**Comment by igwyd** [comment]
In this case the user receives a message "connection lost". We have the bug 74133 for this issue, not fixed yet.
